jax/_src/numpy/lax_numpy.py

```diff
@@ -6397,8 +6400,27 @@ def trim_zeros(filt, trim='fb'):
     >>> jnp.trim_zeros(x)
     Array([2, 0, 1, 4, 3], dtype=int32)
   """
-  filt = core.concrete_or_error(asarray, filt,
+  if isinstance(filt, list):
```

Comment:
This is too specific: we need to deprecate all non-arraylike inputs (for example, tuples will still be accepted here without any warnings).

One way to do this easily is to use `util.check_arraylike` with `emit_warning=True`, though that helper doesn't yet interact with the `deprecations` tools. For now, you could just use that and not do any of the `deprecations.register` stuff.

Reply:
Thanks! Updated as suggested.